packages/twenty-front/src/modules/auth/hooks/useGenerateCaptchaToken.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable @nx/workspace-explicit-boolean-predicates-in-if */
+import { useCallback, useRef } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { captchaProviderState } from '@/client-config/states/captchaProviderState';
+import { CaptchaDriverType } from '~/generated-metadata/graphql';
+
+declare global {
+  interface Window {
+    grecaptcha?: any;
+    turnstile?: any;
+    hcaptcha?: any;
+  }
+}
+
+export const useGenerateCaptchaToken = () => {
+  const captchaProvider = useRecoilValue(captchaProviderState);
+  // eslint-disable-next-line @nx/workspace-no-state-useref
+  const captchaWidgetId = useRef<string | null>(null);
+
+  const generateGoogleRecaptchaToken = useCallback(async (): Promise<
+    string | undefined
+  > => {
+    if (
+      'grecaptcha' in window &&
+      captchaProvider?.provider &&
+      captchaProvider?.siteKey
+    ) {
+      if (captchaWidgetId.current === null) {
+        captchaWidgetId.current = (window.grecaptcha as any).render(
+          'captcha-widget',
+          {
+            sitekey: captchaProvider.siteKey,
+          },
+        );
+      }
+
+      return new Promise((resolve, reject) => {
+        (window.grecaptcha as any)
+          .execute(captchaWidgetId.current, { action: 'submit' })
+          .then((token: string) => resolve(token))
+          .catch((error: any) => reject(error));
+      });
+    }
+  }, [captchaProvider?.provider, captchaProvider?.siteKey]);
+
+  const generateTurnstileCaptchaToken = useCallback(async (): Promise<
```

Comment:
I didn't realize it would be so much work to support different providers sorry!
Do you think you could factor a little bit more to avoid duplicated code here? If not no problem